src/test/tst_attributerange.cpp

```diff
@@ -30,10 +31,632 @@ class TestAttributeRange: public QObject
 private slots:
     void initTestCase() {}
     void cleanupTestCase() {}
+    void tst_bool();
+    void tst_int_range();
+    void tst_double_range();
+    void tst_int_set();
+    void tst_string();
+    void tst_string_set();
+    void tst_double_set();
+    void tst_filepath();
+    void tst_dirpath();
 
+    // Auxiliary function
+    void _tst_value(Value v, Value::Type type);
 };
 
-// TO DO:
+void TestAttributeRange::_tst_value(Value v, Value::Type type)
+{
+    Value::Type t = v.type();
+    if(t == Value::INVALID)  QVERIFY(!v.isValid());
+    else QVERIFY(v.isValid());
+
+    QCOMPARE(t, type);
+
+    switch (type) {
+    case Value::BOOL:
+        QVERIFY(v.isBool());
+        QVERIFY(!v.isInt());
+        QVERIFY(!v.isChar());
+        QVERIFY(!v.isDouble());
+        QVERIFY(!v.isString());
+        break;
+    case Value::INT:
+        QVERIFY(v.isInt());
+        QVERIFY(!v.isBool());
+        QVERIFY(!v.isChar());
+        QVERIFY(!v.isDouble());
+        QVERIFY(!v.isString());
+        break;
+    case Value::DOUBLE:
+        QVERIFY(v.isDouble());
+        QVERIFY(!v.isInt());
+        QVERIFY(!v.isBool());
+        QVERIFY(!v.isChar());
+        QVERIFY(!v.isString());
+        break;
+    case Value::STRING:
+        QVERIFY(v.isString());
+        QVERIFY(!v.isInt());
+        QVERIFY(!v.isBool());
+        QVERIFY(!v.isDouble());
+        QVERIFY(!v.isChar());
+        break;
+    default:
+        QVERIFY(!v.isString());
+        QVERIFY(!v.isInt());
+        QVERIFY(!v.isBool());
+        QVERIFY(!v.isDouble());
+        QVERIFY(!v.isChar());
+        break;
+    }
+}
+
+void TestAttributeRange::tst_bool()
+{
+    AttributeRange* attrRge = AttributeRange::parse(0, "test", "bool");
+
+    // Tests value returned by 'AttributeRange::validate()' for true
+    Value v;
+
+    v = attrRge->validate("true");
+    _tst_value(v, Value::BOOL);
+
+    // Tests value returned by 'AttributeRange::validate()' for false
+    v = attrRge->validate("false");
+    _tst_value(v, Value::BOOL);
+
+    // Tests value returned by 'AttributeRange::validate()' for an invalid string
+     v = attrRge->validate("invalid");
+     _tst_value(v, Value::INVALID);
+
+    // Tests if functions work as expected
+    QVERIFY(attrRge->isValid());
+    QCOMPARE(attrRge->id(), 0);
+    QCOMPARE(attrRge->attrName(), QString("test"));
+    QCOMPARE(attrRge->attrRangeStr(), QString("bool"));
+
+    AttributeRange::Type type = AttributeRange::Bool;
+    QCOMPARE(attrRge->type(), type);
+
+    // Tests min(), max() and rand() functions work as expected
+    // where min and max are boolean values
+    bool min_bool = false;
+    bool max_bool = true;
+
+    // where min and max are integers, casted to boolean type
```

Comment:
The `toX()` methods won't perform any casting. They just convert a `Value` to the type it was built. For instance, if we build a `Value` with a boolean, eg., `Value a(true);`, the only way to get this boolean back is by calling `a.toBool()`. Anything else will and should fail. Thus, please remove the `min_int` and `max_int` tests here.